feat(android): migrate to gradle

jquick-axway opened the pull request:
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27725

**Summary:**
- Now builds with gradle to an "AAR" library.
- Set min Titanium SDK version to `9.0.0`.
- Added `x86_64` architecture support.
- Updated Facebook library version from `5.4.0` to `5.13.0`.
- Removed deprecated `InviteDialog`.
  * No longer works. See: [here](https://developers.facebook.com/docs/archive/docs/app-invites/android)
- Removed module's `onAppCreate()` method.
  * Facebook's SDK initializer was deprecated. Now auto-inits via `ContentProvider`.
- `LoginButton` properties `publishPermissions` and `readPermissions` have been deprecated.
  * Must use `permissions` property instead.

vijaysingh-axway committed:
docs(ios): update apidocs

vijaysingh-axway committed:
fix(ios): moved getters inside proxy to get actually called

jquick-axway committed:
feat(android): migrate to gradle

- Set min Titanium SDK version to 9.0.0.
- Added x86_64 architecture support.
- Updated Facebook library to 5.13.0

hansemannn committed:
chore: replace/remove deprecated api’s

jquick-axway committed:
Merge pull request #1 from hansemannn/TIMOB-27725

chore(android): replace/remove deprecated api’s

jquick-axway committed:
chore(android): improved code formatting

garymathews commented on android/src/facebook/LoginButtonView.java:
```diff
@@ -48,19 +48,19 @@ public LoginButtonView(TiViewProxy proxy)
-				loginButton.setPublishPermissions(Arrays.asList(publishPermissions));
-			}
+			Log.w(TAG,
+				  "The \"publishPermissions\" property has been deprecated in favor of the \"permissions\" property");
```
Logic for `publishPermissions ` and `readPermissions` needs re-adding. Since these changes remove these properties and not deprecate them.

hansemannn replied:
@garymathews They are actually removed in the native SDK already (no-op functions natively), but it can be added again I think. On iOS (merged on master), it's the same behavior as here right now I think.

garymathews replied:
We can just pass to
```Java
loginButton.setPermissions(Arrays.asList(...));
```
But if we can't, then I guess we can just update the log message to state it's been removed.

jquick-axway replied:
Updated PR to merge deprecated `publishPermissions` and `readPermissions` property arrays to the main non-deprecated `permissions` collection/method. Kept the logged warnings. There's slight duplication in the module's Java `propertyChanged()` method, but I think we can live with it for now... until we remove them later.

jquick-axway committed:
chore(android): added temp support for deprecated permission props

- Permission properties "publishPermissions" and "readPermissions" have been deprecated by Facebook.
  * Now routed to LoginButton's main permissions() method, but will log a warning when used.

jquick-axway committed:
chore: updated Jenknins file to build with SDK 9.0.0

sgtcoolguy committed:
chore(package): add @seadub/danger-plugin-titanium-module@0.0.4

sgtcoolguy committed:
chore(package): update lockfile

sgtcoolguy committed:
ci(danger): add module linting to danger.js

sgtcoolguy committed:
build(ios): use sdk 9.0.0.GA to build in xcode
